### ISSUE TITLE
Why use learning_rate=learning_rate*worker_num?

### ISSUE THREAD

**Issue by cklsoft** [horovod/horovod#930]
**Your question:**
Please ask your question here.

I doubt that why use learning_rate=learning_rate*worker_num? What will happen if still use origin learning_rate?

**Comment by hubutui** [comment]
check #241

**Comment by tgaddair** [comment]
Hey @cklsoft, because Horovod's `DistributedOptimizer` averages the gradients across all workers, if you don't increase the learning rate, then you will travel a much smaller distance through your parameter search space each epoch.

For example, suppose your dataset works out to 100 batches per epoch.  If using Horovod you scale up to 10 workers, then each worker will only be processing 10 batches per epoch.  Because the magnitude of the gradients after averaging will be the same in the non-distributed case, the result will be taking 1/10 the number of gradient update steps each epoch.

By scaling the learning rate, you're still taking 1/10 the number of steps, but each step is going to be 10x bigger, so you'll end up walking the same distance each epoch.

**Comment by maxhgerlach** [comment]
Note: When you are using Adam or some other optimizer apart from SGD, it is less clear that the learning rate needs to be scaled.

**Comment by hubutui** [comment]
@maxhgerlach Interesting, could you share more details?

**Comment by maxhgerlach** [comment]
Adam normalizes the gradients, after those have been averaged over the workers via allreduce, in an involved way with first and second moment decay. Only the result of that is multiplied by the overall learning rate factor before it is applied to the weights. This is quite a lot more involved than straight SGD, therefore the motivation for the linear scaling rule as presented in Sec. 2.1 of the [Imagenet in one hour](https://research.fb.com/wp-content/uploads/2017/06/imagenet1kin1h5.pdf) paper no longer applies directly.

I would love to see some empirical results on linear learning rate scaling with Adam in large batches.

**Comment by kaituoxu** [comment]
Hi @maxhgerlach , do you have some empirical results to share? Thx

**Comment by lxiongh** [comment]
> Adam normalizes the gradients, after those have been averaged over the workers via allreduce, in an involved way with first and second moment decay. Only the result of that is multiplied by the overall learning rate factor before it is applied to the weights. This is quite a lot more involved than straight SGD, therefore the motivation for the linear scaling rule as presented in Sec. 2.1 of the [Imagenet in one hour](https://research.fb.com/wp-content/uploads/2017/06/imagenet1kin1h5.pdf) paper no longer applies directly.
> 
> I would love to see some empirical results on linear learning rate scaling with Adam in large batches.

Hi, @maxhgerlach , any empirical results for Adam vs SGD when training in multi-workers?

**Comment by maxhgerlach** [comment]
State of the art language modeling or machine translation papers all use Adam or Adafactor optimizers with really large distributed batches as far as I can tell. SGD does not cut it there.
